exercises/1/webapp/src/components/Home.vue

```diff
@@ -0,0 +1,171 @@
+<template>
+  <div id="app">
+    <div class="news-list">
+      <div class="news-list-wrapper">
+        <h1>{{msg}}</h1>
+        <div class="news-holder" v-for="(newsContent, index) in sortedNewsList" :key="index">
+          <News :newsTitle="newsContent.title" :newsBody="newsContent.body" :newsVotes="newsContent.votes" @updateItem="mewsUpdateMessage" @removeItem="newsRemoveMessage"/>
+        </div>
+      </div>
+      <form @submit.prevent="checkForm">
+        <p v-if="errors.length" class="alert">
+          <b>Please correct the following error(s):</b>
+          <ul>
+            <li v-for="(error, index) in errors" :key="index">{{ error }}</li>
+          </ul>
+        </p>
+        <input v-model="newNews.title" placeholder="Title">
+        <input v-model="newNews.body" placeholder="Body">
+        <button type="submit"> Create </button>
+      </form>
+    </div>
+  </div>
+</template>
+
+<script>
+import News from './News.vue'
+
+export default {
+  name: "Home",
+  props: {
+    msg: {
+      type: String,
+      required: false,
+      default: 'Welcome to Hackernews'
+    },
+    newsTitle: {
+      type: String,
+      required: false,
+      default: ''
+    },
+    newsBody: {
+      type: String,
+      required: false,
+      default: ''
+    }
+  },
+  data() {
+      return {
+        errors: [],
+        newNews: {
+          title : "",
+          body : ""
+        },
+        newsList: [
+          {
+            title : "VueJS",
+            body : "VueJS is nice",
+            votes : 0
+          }, 
+          {
+            title : "Angular",
+            body : "Angular is the best!",
+            votes : 0
+          }, 
+          {
+            title : "Java",
+            body : "java is perfect!",
+            votes : 0
+          }
+        ],
+      }
+  },
+  computed: {
+    sortedNewsList: function() {
+      // Sort by votes
+      function compare(a, b) {
+        if (a.votes < b.votes)
+          return 1
+        if (a.votes > b.votes)
+          return -1
+        return 0
+      }
+      const arrayClone = Array.from(this.newsList)
+      return arrayClone.sort(compare)
+    }
+  },
+  methods: {
+      mewsUpdateMessage(newsTitle, newsVotes, voteChange){
+        const newsId = this.newsList.findIndex((item) => item.title == newsTitle)
+        if (newsId != -1) {
+          console.log("New Update Message from Child (" + newsId + "), Votes: " + (newsVotes + voteChange))
+          this.newsList[newsId].votes = newsVotes + voteChange
+        }
+        else {
+          console.warn("News with title (" + newsTitle + ") not founded in the list of news")
```

Comment:
```suggestion
          console.warn(`News with title (${newsTitle}) not founded in the list of news`)
```

But better remove the debug output altogether:

```suggestion
```